pyvista/core/filters/poly_data.py

```diff
@@ -1138,6 +1145,123 @@ def decimate_pro(
 
         return mesh
 
+    def decimate_polyline(
+        self,
```

Comment:
Let's prepare for additional arguments before the `progress_bar`.
```suggestion
        self,
        *,
```

Reply:
Even though I'm +1 for this style in general, I think we should have a standard style for the whole project. It will be surprising to users if only some filters have it. What do you think about just making `progress_bar` keyword only?  This has lower downsides.

Reply:
Using `*` for kwargs is already implemented in a lot of new code, so there is already an inconsistency there. I don't think using it (or not) here will make-or-break this though, so I don't think using `*` (or not) should block this PR.

But, IMO the benefits of keyword-only args far outweigh this inconsistency. I would even consider the use of any positional `bool` arg as a bad practice in general which should be avoided. It's not clear if or when keyword-only args will be enforced for legacy code, but in the mean time I don't think we should go out of way to avoid using it in new code.

Reply:
Ok I'm +1, but the suggestion needs to be modified as `reduction` should not be keyword only.  Will do this soon.

Reply:
+1 from me and I would love to standardize this across the entire code base